ntp-daemon/src/ipfilter.rs

```diff
@@ -143,25 +137,33 @@ impl BitTree {
                 }
             }
         }
-        // compensate for incorrectly marked outsets due to overcoverage
+
+        // the outset should not contain anything that is included in the inset
+        // (this can happen due to overcoverage)
         node.outset &= !node.inset;
 
+        // bitmap of subsegments for which we have a decision
         let known_bitmap = node.inset | node.outset;
-        // allocate nodes
-        self.nodes.resize(
-            self.nodes.len() + (!known_bitmap).count_ones() as usize,
-            TreeNode::default(),
-        );
+
+        // allocate additional empty nodes
+        let unknown_count = known_bitmap.count_zeros() as usize;
+        self.nodes
+            .extend(std::iter::repeat(TreeNode::default()).take(unknown_count));
```

Comment:
I think `extend` captures the idea here better than `resize`

Reply:
Hmm, think this might lead to worse performance in the generated code, but since this data structure already chooses to sacrifice performance at create time and this will be at most 16 nodes, I think the tradeoff is worth it.